src/components/tables/TableHeadingRowConnected.tsx

```diff
@@ -3,21 +3,27 @@ import {connect} from 'react-redux';
 import * as _ from 'underscore';
 import {IReactVaporState} from '../../ReactVapor';
 import {IDispatch, ReduxUtils} from '../../utils/ReduxUtils';
+import {updateSelectedRows} from './TableActions';
 import {ITableHeadingRowOwnProps, ITableHeadingRowProps, TableHeadingRow} from './TableHeadingRow';
+import {ITableState} from './TableReducers';
 import {addRow, removeRow, selectRow} from './TableRowActions';
 import {ITableRowState} from './TableRowReducers';
 
 const mapStateToProps = (state: IReactVaporState, ownProps: ITableHeadingRowOwnProps) => {
     const item: ITableRowState = _.findWhere(state.rows, {id: ownProps.id});
+    const table: ITableState = state.tables && state.tables[ownProps.tableId];
 
     return {
         opened: item && item.opened,
-        selected: item && item.selected,
+        selected: table && table.data.selectedIds && !!table.data.selectedIds.length && table.data.selectedIds.indexOf(ownProps.rowId) !== -1,

```

Comment:
Not sure if it can happen, but if `table.data` is undefined, `table.data.selectedIds` will throw an error.